src/data_conn.c

```diff
@@ -210,22 +215,20 @@ handle_data_conn_error(replica_t *r)
 	spec = r->spec;
 
 	TAILQ_FOREACH(r1, &(spec->rq), r_next)
-		if (r1 == r)
+		if (r1 == r) {
+			update_spec_status= 1;
 			break;
-
-	if (r1 == NULL) {
```

Comment:
this condition is required here.
As comment from satbir, there are corner cases (update_replica_entry got timed out to check for some socket, but, it got created later) where 'r' may not be part of spec_rq list